### PR TITLE
zephyr: secondary cores power down

### PR DESCRIPTION
**Pull request by tmleman**
Adding implementation of cpu cores disabling function with usage of a zephyr power manager.
Refactorization of a existing code and moving it to the target location (out of the zephyr wrapper).

### CODE REVIEW

File: src/platform/Kconfig [hunk 1]
```diff
@@ -532,4 +532,11 @@ config SOF_STACK_SIZE
 	  and might be useful when creating more complex audio processing
 	  components.
 
+config SECONDARY_CORE_DISABLING_TIMEOUT
+	int
+	default 1000
+	depends on MULTICORE
+	help
+	  Count of loops used when waiting for secondary core to enter OFF state.
```
Comment (lyakh) [inline src/platform/Kconfig:540]: maybe milliseconds?

Comment (tmleman) [reply src/platform/Kconfig:540]: @lyakh  Only existing function that I have found with timeout functionality is _**void wait_delay_ms(uint64_t ms)**_. In that use case I don't want to wait the whole time if its not necessary. Loop count with state check was the simplest solution.

Comment (lyakh) [reply src/platform/Kconfig:540]: @tmleman you can use that loop internally, no problem with that. But this Kconfig parameter doesn't explain anything to the user - how long is each loop iteration? You can even literally keep your implementation the same but here don't let the user set a loop count, but a maximum wait time (if that needs to be configurable at all). Make it "maximum wait time in ms" and then calculate your loop count from that.

**Comment by lgirdwood** [comment]
@tmleman is this really draft (WIP) PR ? Looks like it should be normal PR.

**Pull request closed without merging**